tests/unit/test_commands.py

```diff
@@ -120,3 +122,35 @@ def is_requirement_command(command: Command) -> bool:
         return isinstance(command, RequirementCommand)
 
     check_commands(is_requirement_command, ["download", "install", "wheel"])
+
+
+@pytest.mark.parametrize(
+    "command", ["cache", "check", "config", "freeze", "hash", "help", "inspect", "show"]
+)
+def test_no_network_imports(command: str) -> None:
```

Comment:
This test is admittedly janky, I would appreciate ideas for preventing regressions that are cleaner :)

Reply:
A more robust approach that doesn't require using the internals would be to use [runpy](https://docs.python.org/3/library/runpy.html#module-runpy). You could do something like:

```py
import runpy
import sys

sys.argv[1:] = [{command}, "--help"]

try:
    runpy.run_module("pip", alter_sys=True, run_name="__main__")
finally:
    with open({file!r}, "w") as f:
        print(list(sys.modules.keys()), file=f)
```

and then have `file` and `command` templated in as f-strings (file being under `tmp_path` fixture's folder would be cleanest, I think).

Also this test creates a bunch of subprocesses, and should live under integration tests for that reason.

Reply:
Ah, that's much cleaner. Thanks a ton!